libcaja-private/caja-dnd.c

```diff
@@ -454,8 +467,8 @@ caja_drag_default_drop_action_for_icons (GdkDragContext *context,
     }
 
     dropped_uri = ((CajaDragSelectionItem *)items->data)->uri;
-    dropped_file = caja_file_get_existing_by_uri (dropped_uri);
-    target_file = caja_file_get_existing_by_uri (target_uri_string);
+    dropped_file = caja_file_get_by_uri (dropped_uri);
+    target_file = caja_file_get_by_uri (target_uri_string);
```

Comment:
I love the approach of checking the file system id to see whether the two files are in the same file system.
However, I don't think it's right adding the files to the hash table here if they don't exist yet.

Reply:
I have updated the code. Thank you.